ark/utils/data_utils_test.py

```diff
@@ -261,91 +71,6 @@ def test_generate_deepcell_input():
         assert np.array_equal(fov2[:, :, 0], data_xr.loc['fov2', :, :, 'nuc2'].values)
 
 
-def test_combine_xarrays():
-    # test combining along points axis
-    fov_ids, chan_ids = test_utils.gen_fov_chan_names(num_fovs=5, num_chans=3)
-
-    base_xr = test_utils.make_images_xarray(tif_data=None, fov_ids=fov_ids, channel_names=chan_ids)
-
-    test_xr = data_utils.combine_xarrays((base_xr[:3, :, :, :], base_xr[3:, :, :, :]), axis=0)
-    assert test_xr.equals(base_xr)
-
-    # test combining along channels axis
-    fov_ids, chan_ids = test_utils.gen_fov_chan_names(num_fovs=3, num_chans=5)
-
-    base_xr = test_utils.make_images_xarray(tif_data=None, fov_ids=fov_ids, channel_names=chan_ids)
-
-    test_xr = data_utils.combine_xarrays((base_xr[:, :, :, :3], base_xr[:, :, :, 3:]), axis=-1)
-    assert test_xr.equals(base_xr)
-
-
-def test_crop_helper():
-    # test crops that divide evenly
-    crop_input = np.zeros((4, 256, 256, 4))
-    crop_size = 64
-
-    cropped = data_utils.crop_helper(crop_input, crop_size)
-    num_crops = crop_input.shape[0] * \
-        (crop_input.shape[1] // crop_size) * (crop_input.shape[2] // crop_size)
-    assert np.array_equal(cropped.shape, (num_crops, crop_size, crop_size, crop_input.shape[3]))
-
-    # test crops that don't divide evenly
-    crop_input = np.zeros((4, 256, 256, 4))
-    crop_size = 56
-
-    cropped = data_utils.crop_helper(crop_input, crop_size)
-    num_crops = crop_input.shape[0] * \
-        ((crop_input.shape[1] + crop_size) // crop_size) * \
-        ((crop_input.shape[2] + crop_size) // crop_size)
-    assert np.array_equal(cropped.shape, (num_crops, crop_size, crop_size, crop_input.shape[3]))
-
-
-def test_crop_image_stack():
-    # test without overlap (stride_fraction = 1)
-    crop_input = np.zeros((4, 256, 256, 4), dtype="int16")
-    crop_size = 64
-    stride_fraction = 1
-
-    cropped = data_utils.crop_image_stack(crop_input, crop_size, stride_fraction)
-    num_crops = crop_input.shape[0] * (crop_input.shape[1] // crop_size) * \
-        (crop_input.shape[2] // crop_size) * (1 / stride_fraction)
-
-    assert np.array_equal(cropped.shape, (num_crops, crop_size, crop_size, crop_input.shape[3]))
-
-    # test with overlap
-    crop_input = np.zeros((4, 256, 256, 4), dtype="int16")
-    crop_size = 64
-    stride_fraction = 0.25
-
-    cropped = data_utils.crop_image_stack(crop_input, crop_size, stride_fraction)
-    num_crops = crop_input.shape[0] * (crop_input.shape[1] // crop_size) * \
-        (crop_input.shape[2] // crop_size) * (1 / stride_fraction) * (1 / stride_fraction)
-
-    assert np.array_equal(cropped.shape, (num_crops, crop_size, crop_size, crop_input.shape[3]))
```

Comment:
used to test with the best 😭

Reply:
Ditto

Reply:
Yeah RIP, I saw you added a bunch of testing for these functions in one of your PRs